spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
+        self.main = main
+
+    def get_pythonpath(self):
+        """
+        Return directories to be added to the Python path.
+
+        Use Python path from Spyder. Overrides function in base class.
+
+        Returns
+        -------
+        list of str
+        """
+        return self.main.get_spyder_pythonpath()
+
+    def get_default_config(self):
+        """
+        Return configuration which is proposed when current config is invalid.
+
+        Propose to use directory of current file as working directory for
+        testing.
+        """
+        filename = self.main.editor.get_current_filename()
+        dirname = os.path.dirname(filename)
+        return Config(wdir=dirname)
+
+
+class UnitTestPlugin(SpyderPluginWidget):
+    """Spyder plugin for unit testing."""
 
     CONF_SECTION = 'unittest'
-    edit_goto = Signal(str, int, str)
 
     def __init__(self, parent=None):
-        """Unit testing."""
-        UnitTestWidget.__init__(self, parent=parent)
-        SpyderPluginMixin.__init__(self, parent)
+        """Initialize plugin and corresponding widget."""
+        if spyder.version_info[0] < 4 and PYQT5:
```

Comment:
Unfortunately this did not work for me with the `3.x` branch. I got the following error:

```
Traceback (most recent call last):
  File "[...]/spyder/app/mainwindow.py", line 1044, in setup
    plugin.register_plugin()
  File "[...]/spyder_unittest/unittestplugin.py", line 119, in register_plugin
    self.main.add_dockwidget(self)
AttributeError: 'NoneType' object has no attribute 'add_dockwidget'
```
I'll have a look and see whether this is caused by something I'm doing wrong.